FWCore/ParameterSet/python/Config.py

```diff
@@ -1156,19 +1182,49 @@ def _insertPaths(self, processPSet, nodeVisitor):
             for name in self.endpaths_():
                 scheduledPaths.append(name)
                 endpaths.append(name)
+            for name in self.finalpaths_():
+                finalpaths.append(name)
         else:
             for path in self.schedule_():
                 pathname = path.label_()
-                scheduledPaths.append(pathname)
                 if pathname in self.endpaths_():
                     endpaths.append(pathname)
+                    scheduledPaths.append(pathname)
+                elif pathname in self.finalpaths_():
+                    finalpaths.append(pathname)
+                    if 1 == len(finalpaths):
+                      scheduledPaths.append("@finalPath")
```

Comment:
This looks like duplication of
```py
        endPathWithFinalPathModulesName ="@finalPath"
        ...
        if finalpaths:
          ...
          scheduledPaths.append(endPathWithFinalPathModulesName)
```
below.

Reply:
fixed